common/include/pcl/point_cloud.h

```diff
@@ -506,22 +545,60 @@ namespace pcl
         height = 1;
       }
 
+      /**
+       * \brief Replaces the points with `new_width * new_height` copies of `value`
+       * \param[in] new_width new width of the point cloud
+       * \param[in] new_height new height of the point cloud
+       */
+      void
+      assign(index_t new_width, index_t new_height, const PointT& value)
+      {
+        points.assign(new_width * new_height, value);
+        width = new_width;
+        height = new_height;
+      }
+
       /**
        * \brief Replaces the points with copies of those in the range `[first, last)`
        * \details The behavior is undefined if either argument is an iterator into
        * `*this`
        * \note This breaks the organized structure of the cloud by setting the height to
        * 1!
        */
-      template <class InputIt>
+      template <class InputIterator>
       void
-      assign(InputIt first, InputIt last)
+      assign(InputIterator first, InputIterator last)
```

Comment:
Another option could be to add a parameter to this with default value, like this:
`assign(InputIterator first, InputIterator last, uindex_t new_height=1)` and `width = size() / height`
That way we would have fewer functions overall